File: src/assets/usage-examples-code/timers/stop_timer/stop_timer-1-basic-usage.cpp
#include <iostream>
#include "splashkit.h"

int main()
{
    timer myTimer = create_timer("my_timer");
    start_timer(myTimer);
    std::cout << "Timer started." << std::endl;
    delay(2000); 

    // Stop the timer
    stop_timer(myTimer);
    std::cout << "Timer stopped." << std::endl;

    delay(3000); 
    reset_timer(myTimer);
    std::cout << "Timer reset." << std::endl;
    delay(1000);
    start_timer(myTimer);
    std::cout << "Timer restarted." << std::endl;
    delay(500); 
    free_timer(myTimer);
    std::cout << "Timer resource freed." << std::endl;

    return 0;
}

File: src/assets/usage-examples-code/timers/stop_timer/stop_timer-1-basic-usage.cs
using System;
using SplashKitSDK; // Import SplashKit namespace

using SplashKitTimer = SplashKitSDK.Timer; // Alias for SplashKit's Timer class

class Program
{
    static void Main(string[] args)
    {
        SplashKitTimer myTimer = new SplashKitTimer("my_timer");
        myTimer.Start();
        Console.WriteLine("Timer started.");
        System.Threading.Thread.Sleep(2000); 

        // Stop the timer
        myTimer.Stop();
        Console.WriteLine("Timer stopped.");


        System.Threading.Thread.Sleep(3000); 
        myTimer.Reset();
        Console.WriteLine("Timer reset.");
        System.Threading.Thread.Sleep(1000); 
        myTimer.Start();
        Console.WriteLine("Timer restarted.");
        System.Threading.Thread.Sleep(500); 
        myTimer.Dispose();
        Console.WriteLine("Timer resource disposed.");
    }
}
